biloba/test/test_service.py

```diff
@@ -462,3 +462,53 @@ def test_biloba_config(self):
         svc = service.ConfigurableService(cfg)
 
         self.assertIs(svc.config, cfg)
+
+
+class AutospawnTestCase(unittest.TestCase):
+    """
+    Tests for `service.autospawn`.
+    """
+
+    def test_started(self):
+        class MyService(service.Service):
+            @service.autospawn
+            def my_func(self):
+                self.emit('autospawn')
+
+        my_service = MyService()
+        self.executed = False
+
+        @my_service.on('autospawn')
+        def on_autospawn():
+            self.assertTrue(my_service.started)
+            self.executed = True
+
+        with my_service:
+            ret = my_service.my_func()
+
+            self.assertIsNone(ret)
+
+        self.assertTrue(self.executed)
+
+    def test_not_started(self):
+        class MyService(service.Service):
+            @service.autospawn
+            def my_func(self):
+                self.emit('autospawn')
+
+        my_service = MyService()
+        self.executed = False
+
+        @my_service.on('autospawn')
+        def on_autospawn():
+            self.assertTrue(my_service.started)
+            self.executed = True
+
+        ret = my_service.my_func()
+
+        self.assertIsNone(ret)
+
+        self.assertFalse(self.executed)
+
+        with my_service:
+            self.assertTrue(self.executed)
```

Comment:
Should you add another test to make sure it executes once the service is started?